rust/eos-rs/src/api/dungeon_mode/monster.rs

```diff
@@ -1,55 +1,122 @@
+use crate::api::abilities::AbilityId;
 use crate::api::dungeon_mode::*;
+use crate::api::iq::IqSkillId;
+use crate::api::items::{ExclusiveItemEffectId, ItemId};
+use crate::api::monsters::MonsterSpeciesId;
 use crate::api::moves::*;
-use crate::api::objects::*;
-use crate::ffi::type_id::Type;
+use crate::api::types::MonsterTypeId;
+use core::ops::{Deref, DerefMut};
 use fixed::types::I24F8;
 
-/// Extension trait for [`DungeonMonsterRef`] (read operations).
+/// Reference type for the info struct for [`DungeonEntity`] objects that are monsters.
+///
+/// For methods, see the [`DungeonMonsterRead`] and [`DungeonMonsterWrite`] traits.
```

Comment:
I think you mixed these up. `DungeonMonsterRef` should refer only to `DungeonMonsterRead`, and `DungeonMonsterMut` should refer to both `DungeonMonsterRead` and `DungeonMonsterWrite`.